android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt

```diff
@@ -131,6 +133,69 @@ class AirshipPlugin : MethodCallHandler {
         result.success(null)
     }
 
+    private fun addEvent(call: MethodCall, result: Result) {
+        val eventMap = call.arguments as HashMap<*, *>
+
+        val eventName = eventMap[CustomEvent.EVENT_NAME] as String? ?: run {
+            result.success(false)
+            return
+        }
+
+        val event = CustomEvent.Builder(eventName).apply {
+            (eventMap[CustomEvent.EVENT_VALUE] as Int?)?.let {
+                this.setEventValue(it)
+            }
+
+            (eventMap[CustomEvent.PROPERTIES] as HashMap<String, Any>?)?.let {
+                parseProperties(it, this)
+            }
+
+            (eventMap[CustomEvent.TRANSACTION_ID] as String?)?.let {
+                this.setTransactionId(it)
+            }
+
+            val interactionId = eventMap[CustomEvent.INTERACTION_ID] as String?
+            val interactionType = eventMap[CustomEvent.INTERACTION_TYPE] as String?
+
+            if (interactionId != null && interactionType != null) {
+                this.setInteraction(interactionType, interactionId)
+            }
+        }.build()
+
+        if (event.isValid) {
+            event.track()
+            result.success(true)
+        } else {
+            result.success(false)
+        }
+    }
+
+    private fun parseProperties(map:HashMap<String, Any>, builder:CustomEvent.Builder) {
+        for ((key, value) in map) {
+            if (value is Int) {
+                builder.addProperty(key, value)
+                continue
+            }
+
+            if (value is Boolean) {
+                builder.addProperty(key, value)
+                continue
+            }
+
+            if (value is String) {
+                builder.addProperty(key, value)
+                continue
+            }
```

Comment:
Might be a coding style thing, but can these three `if`s be compressed into one with three `OR`s?

Reply:
explained in person